### PR TITLE
Refactor heads to support multiple heads

### LINKED ISSUE #219: multi-head design
hi，if I would like to use multi-head for the keypoint prediction, there is only one key word("keypoint_head") in the model config, how can I realize it? Could you give me some advice?  Hope for your reply! @jin-s13

### PR DESCRIPTION
**Pull request by jin-s13**
fix #219 

Migration note :  

Config files need to be changed.

1. rename `loss_pose` to `loss_keypoint`.

2. move `loss_keypoint` from **model** settings to **keypoint_head** settings.

For example, in `configs/top_down/hrnet/coco/hrnet_w32_coco_256x192.py`. 

from 

```
keypoint_head=dict(
        type='TopDownSimpleHead',
        in_channels=32,
        out_channels=channel_cfg['num_output_channels'],
        num_deconv_layers=0,
        extra=dict(final_conv_kernel=1, ),
    ),
    train_cfg=dict(),
    test_cfg=dict(
        flip_test=True,
        post_process='default',
        shift_heatmap=True,
        modulate_kernel=11),
    loss_pose=dict(type='JointsMSELoss', use_target_weight=True))
```
to

```
keypoint_head=dict(
        type='TopDownSimpleHead',
        in_channels=32,
        out_channels=channel_cfg['num_output_channels'],
        num_deconv_layers=0,
        extra=dict(final_conv_kernel=1, ),
        loss_keypoint=dict(type='JointsMSELoss', use_target_weight=True)),
    train_cfg=dict(),
    test_cfg=dict(
        flip_test=True,
        post_process='default',
        shift_heatmap=True,
        modulate_kernel=11))
```

### CODE REVIEW

**Review by wusize — APPROVED** [review]

File: mmpose/models/keypoint_heads/top_down_base_head.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        """Inference function."""
+
+    def decode_keypoints(self, img_metas, output_heatmap):
+        batch_size = len(img_metas)
+
+        if batch_size > 1:
+            assert 'bbox_id' in img_metas[0]
+            bbox_ids = []
+        else:
+            bbox_ids = None
```
Comment (wusize) [inline mmpose/models/keypoint_heads/top_down_base_head.py:44]: bbox_id is set as None only when 'bbox_id' is not in img_metas[0].

**Comment by innerlee** [comment]
Give a clear migration note on the first post. e.g., https://github.com/open-mmlab/mmpose/pull/288

File: mmpose/models/detectors/bottom_up.py [hunk 2]
```diff
@@ -26,16 +25,14 @@ class BottomUp(BasePose):
...
-        loss_pose (dict): Config for loss. Default: None.
     """
 
     def __init__(self,
                  backbone,
                  keypoint_head=None,
                  train_cfg=None,
                  test_cfg=None,
-                 pretrained=None,
-                 loss_pose=None):
```
Comment (innerlee) [inline mmpose/models/detectors/bottom_up.py:38]: For better migration experience, we can keep `loss_pose` here, and raise an error if it is not None, and give the link to this pr in the error message.

**Review by innerlee — APPROVED** [review]

**Comment by innerlee** [comment]
Two comments.

**Pull request merged**